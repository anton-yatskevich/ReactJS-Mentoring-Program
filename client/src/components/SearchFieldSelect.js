import React from 'react';
import PropTypes from 'prop-types';

import Button from './Button';

const SearchFieldSelect = ({ onChangeSearchField, searchField }) => (
    <div className="search-panel__field-select">
        <span className="search-panel__field-select-title">Search by</span>
        <Button onClickHandler={onChangeSearchField} value="title" textValue="title" searchField={searchField} />
        <Button onClickHandler={onChangeSearchField} value="genres" textValue="genre" searchField={searchField} />
    </div>
);

SearchFieldSelect.propTypes = {
    onChangeSearchField: PropTypes.func,
    searchField: PropTypes.string
};

SearchFieldSelect.defaultProps = {
    onChangeSearchField: null,
    searchField: ''
};

export default SearchFieldSelect;
