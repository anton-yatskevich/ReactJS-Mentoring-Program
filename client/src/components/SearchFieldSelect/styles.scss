.search-field-select {
    &__field-select-title {
        text-transform: uppercase;
        font-size: 14px;
    }

	&__button {
		padding: 5px 20px;
		font-size: 14px;
		text-transform: uppercase;
		background-color: rgb(173, 173, 173);
		border: none;
		color: white;
		text-decoration: none;
		margin-left: 15px;
		transition: all 0.5;
		cursor: pointer;

		&--active {
			background-color: rgb(160, 0, 0);
		}
	}
}