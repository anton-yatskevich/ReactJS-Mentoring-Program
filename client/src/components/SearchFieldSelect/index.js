import React from 'react';
import PropTypes from 'prop-types';
import noop from 'lodash.noop';
import Button from '../Button';
import './styles.scss';

const SearchFieldSelect = ({ setSearchField, searchField }) => (
    <div className="search-field-select">
        <span className="search-field-select__title">Search by</span>
        <Button onClickHandler={setSearchField} value="title" textValue="title" isActive={searchField === 'title'} className="search-field-select__button" />
        <Button onClickHandler={setSearchField} value="genres" textValue="genre" isActive={searchField === 'genres'} className="search-field-select__button" />
    </div>
);

SearchFieldSelect.propTypes = {
    setSearchField: PropTypes.func,
    searchField: PropTypes.string
};

SearchFieldSelect.defaultProps = {
    setSearchField: noop,
    searchField: ''
};

export default SearchFieldSelect;
