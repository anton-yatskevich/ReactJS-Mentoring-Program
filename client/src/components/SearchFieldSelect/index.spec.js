import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import SearchFieldSelect from './index';

describe('SearchFieldSelect component', () => {
    it('should be rendered with default props', () => {
        const SearchFieldSelectComponent = shallow(<SearchFieldSelect />);
        expect(toJson(SearchFieldSelectComponent)).toMatchSnapshot();
    });
});
