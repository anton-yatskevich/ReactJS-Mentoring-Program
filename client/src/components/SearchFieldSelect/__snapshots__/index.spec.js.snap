// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFieldSelect component should be rendered with default props 1`] = `
<div
  className="search-field-select"
>
  <span
    className="search-field-select__title"
  >
    Search by
  </span>
  <Button
    className="search-field-select__button"
    isActive={false}
    onClickHandler={[Function]}
    textValue="title"
    value="title"
  />
  <Button
    className="search-field-select__button"
    isActive={false}
    onClickHandler={[Function]}
    textValue="genre"
    value="genres"
  />
</div>
`;
